### PR TITLE
App router localization

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Support app router i18n

Server-side translations (test/page) and client-side (test/ClientComponent)

Based on https://i18nexus.com/tutorials/nextjs/react-i18next

To simplify the setup, we always load all namespaces without picking only needed ones

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3923?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #3894** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3894?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3901** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3901?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #3879** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3879?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #3904** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3904?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #3905** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3905?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #3906** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3906?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #3908** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3908?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #3909** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3909?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #3893** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3893?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #3923** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3923?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3923?utm_source=stack-comment).

File: apps/store/src/app/[locale]/i18n.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+const allNamespaces = [
+  'common',
+  'bankid',
+  'carDealership',
+  'cart',
+  'checkout',
+  'contact-us',
+  'memberArea',
+  'purchase-form',
+  'widgetΩ',
```
Comment (gustaveen) [inline apps/store/src/app/[locale]/i18n.ts:18]: Typo? 
```suggestion
  'widget',
```

File: apps/store/src/app/[locale]/i18n.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  async read(language, namespace, callback) {
+    try {
+      const resource = await import(`../../../public/locales/${language}/${namespace}.json`)
+      callback(null, resource)
+    } catch (err: unknown) {
+      callback(err as CallbackError, null)
+    }
+  },
+  init: () => {},
+}
```
Comment (gustaveen) [inline apps/store/src/app/[locale]/i18n.ts:69]: Saw that the tutorial used this package for this https://github.com/i18next/i18next-resources-to-backend

Was it overkill for our use case?

File: apps/store/src/app/[locale]/test/page.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { initTranslationsServerSide } from '../i18n'
+import { ClientComponent } from './ClientComponent'
+import { wrapper } from './styles.css'
+
+type LocalizedPageProps<P = unknown> = P & {
+  params: { locale: RoutingLocale }
+}
+
+const Page = async (props: LocalizedPageProps) => {
+  const { t } = await initTranslationsServerSide(props.params.locale)
```
Comment (gustaveen) [inline apps/store/src/app/[locale]/test/page.tsx:11]: So this loads all name spaces?

**Review by gustaveen — APPROVED** [review]
Good job! 🙌 Just saw a typo

File: apps/store/src/app/[locale]/i18n.ts (hunk 1 above)
Comment (alebedev) [reply apps/store/src/app/[locale]/i18n.ts:18]: Good catch!

File: apps/store/src/app/[locale]/test/page.tsx (hunk 3 above)
Comment (alebedev) [reply apps/store/src/app/[locale]/test/page.tsx:11]: Yes, all by default.  It would be useful to only load some subset, but this is a future improvement

**Pull request merged**